src/android/CodePush.java

```diff
@@ -124,32 +137,36 @@ private boolean execInstall(CordovaArgs args, CallbackContext callbackContext) {
         return true;
     }
 
-    private void markUpdateAndStartTimer(int updateSuccessTimeoutInMillis) {
-    /* this flag will clear when reloading the plugin */
-        this.didUpdate = true;
-
-        if (updateSuccessTimeoutInMillis > 0) {
-            startRollbackTimeout(updateSuccessTimeoutInMillis);
-        } else {
-            cleanOldPackageSilently();
+    private boolean execRestartApplication(CordovaArgs args, CallbackContext callbackContext) {
+        try {
+            /* check if we have a deployed package already */
+            CodePushPackageMetadata deployedPackageMetadata = this.codePushPackageManager.getCurrentPackageMetadata();
+            if (deployedPackageMetadata != null) {
+                callbackContext.success();
+                didStartApp = false;
+                onStart();
+            } else {
+                callbackContext.error("The application has no installed updates.");
```

Comment:
It seems like this will not actually restart the app if there is no CodePush installed package. Perhaps this should be documented as it might cause confusion, and also this behaviour should be described in the comments here:
https://github.com/Microsoft/cordova-plugin-code-push/pull/53/files#diff-cdca2bab8aff8d52c97a48f83b4229d0R51